mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -1342,21 +1369,53 @@ std::string CppEmitter::createMemberAccess(emitc::MemberOfPtrOp op) {
 }
 
 void CppEmitter::cacheDeferredOpResult(Value value, StringRef str) {
-  if (!valueMapper.count(value))
+  if (!valueMapper.count(value)) {
     valueMapper.insert(value, str.str());
+  }
 }
 
 /// Return the existing or a new name for a Value.
 StringRef CppEmitter::getOrCreateName(Value val) {
+  // Check whether value belongs to a loop induction variable
+  if (loopInductionVarMapper.count(val)) {
+    return *loopInductionVarMapper.begin(val);
+  }
+
   if (!valueMapper.count(val)) {
     assert(!hasDeferredEmission(val.getDefiningOp()) &&
            "cacheDeferredOpResult should have been called on this value, "
            "update the emitOperation function.");
-    valueMapper.insert(val, formatv("v{0}", ++valueInScopeCount.top()));
+
+    valueMapper.insert(val,
+                       formatv("v{0}", ++valueInScopeCount.top() + valueCount));
   }
   return *valueMapper.begin(val);
 }
 
+/// Return the existing or a new name for a loop induction variable Value.
+/// Loop induction variables follow natural naming: i, j, k,...
+StringRef CppEmitter::getOrCreateName(emitc::ForOp forOp) {
+  Value val = forOp.getInductionVar();
+
+  if (!loopInductionVarMapper.count(val)) {
+
+    int64_t identifier = loopNestingLevel++;
+
+    char range = 'z' - 'i';
+    if (identifier >= 0 && identifier <= range) {
```

Comment:
nit: Would this work? When can identifier be < 0?
```suggestion
    int64_t identifier = 'i' + loopNestingLevel++;

    if (identifier >= 'i' && identifier <= 'z') {
```